src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/ICustomValidationProcessor.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.juddata.camel.processor;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import uk.gov.hmcts.reform.data.ingestion.camel.validator.JsrValidatorInitializer;
+import uk.gov.hmcts.reform.juddata.camel.binder.JudicialOfficeAppointment;
+import uk.gov.hmcts.reform.juddata.camel.binder.JudicialOfficeAuthorisation;
+import uk.gov.hmcts.reform.juddata.camel.binder.JudicialUserProfile;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static java.util.Objects.nonNull;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.BooleanUtils.isFalse;
+import static org.springframework.util.CollectionUtils.isEmpty;
+import static uk.gov.hmcts.reform.juddata.camel.util.JrdConstants.INVALID_JSR_PARENT_ROW;
+import static uk.gov.hmcts.reform.juddata.camel.util.JrdMappingConstants.ELINKS_ID;
+
+
+public interface ICustomValidationProcessor<T> {
+
+    @Slf4j
+    final class LogHolder {
+    }
+
+    default void filterInvalidUserProfileRecords(List<T> filteredChildren,
+                                                 List<JudicialUserProfile> invalidJudicialUserProfileRecords,
+                                                 JsrValidatorInitializer<T> jsrValidatorInitializer,
+                                                 Exchange exchange, String logComponentName) {
+        Type mySuperclass = getType();
+        List<String> invalidElinks = new ArrayList<>();
+        if (nonNull(invalidJudicialUserProfileRecords)) {
+
+            invalidJudicialUserProfileRecords.forEach(invalidRecords -> {
+                //Remove invalid appointment for user profile and add to invalidElinks List
+                boolean filteredRecord;
+                if (((Class) mySuperclass).getCanonicalName().equals(JudicialOfficeAppointment
+                    .class.getCanonicalName())) {
+                    filteredRecord = filteredChildren.removeIf(filterInvalidUserProfAppointment ->
+                        ((JudicialOfficeAppointment) filterInvalidUserProfAppointment).getElinksId()
+                            .equalsIgnoreCase(invalidRecords.getElinksId()));
+                } else {
+                    filteredRecord = filteredChildren.removeIf(filterInvalidUserProfAppointment ->
```

Comment:
rename filterInvalidUserProfAppointment to something like authorisation for better readability